specs/merge/beacon-chain.md

```diff
@@ -196,50 +188,45 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
     process_operations(state, block.body)
-    # Pre-merge, skip execution payload processing
-    if is_execution_enabled(state, block):
+    if is_execution_enabled(state, block.body):
         process_execution_payload(state, block.body.execution_payload, EXECUTION_ENGINE)  # [New in Merge]
 ```
 
-#### Execution payload processing
+### Execution payload processing
 
-##### `process_execution_payload`
+#### `process_execution_payload`
 
 ```python
-def process_execution_payload(state: BeaconState,
-                              execution_payload: ExecutionPayload,
-                              execution_engine: ExecutionEngine) -> None:
-    """
-    Note: This function is designed to be able to be run in parallel with the other `process_block` sub-functions
-    """
-    if is_transition_completed(state):
-        assert execution_payload.parent_hash == state.latest_execution_payload_header.block_hash
-        assert execution_payload.number == state.latest_execution_payload_header.number + 1
-
-    assert execution_payload.timestamp == compute_time_at_slot(state, state.slot)
-
-    assert execution_engine.new_block(execution_payload)
-
+def process_execution_payload(state: BeaconState, payload: ExecutionPayload, execution_engine: ExecutionEngine) -> None:
+    # Verify consistency of the parent hash and block number
+    if is_merge_complete(state):
+        assert payload.parent_hash == state.latest_execution_payload_header.block_hash
+        assert payload.block_number == state.latest_execution_payload_header.block_number + uint64(1)
+    # Verify timestamp
+    assert payload.timestamp == compute_timestamp_at_slot(state, state.slot)
+    # Verify the execution payload is valid
+    assert execution_engine.on_payload(payload)
+    # Cache execution payload
     state.latest_execution_payload_header = ExecutionPayloadHeader(
-        block_hash=execution_payload.block_hash,
-        parent_hash=execution_payload.parent_hash,
-        coinbase=execution_payload.coinbase,
-        state_root=execution_payload.state_root,
-        number=execution_payload.number,
-        gas_limit=execution_payload.gas_limit,
-        gas_used=execution_payload.gas_used,
-        timestamp=execution_payload.timestamp,
-        receipt_root=execution_payload.receipt_root,
-        logs_bloom=execution_payload.logs_bloom,
-        transactions_root=hash_tree_root(execution_payload.transactions),
+        parent_hash=payload.parent_hash,
+        coinbase=payload.coinbase,
+        state_root=payload.state_root,
+        logs_bloom=payload.logs_bloom,
+        receipt_root=payload.receipt_root,
+        block_number=payload.block_number,
+        gas_limit=payload.gas_limit,
+        gas_used=payload.gas_used,
+        timestamp=payload.timestamp,
+        block_hash=payload.block_hash,
+        transactions_root=hash_tree_root(payload.transactions),
     )
 ```
 
-## Initialize state for pure Merge testnets and test vectors
+## Testing
```

Comment:
This isn't just for testing, imo. It's also for pure merge networks which could be other "mainnets"

but.. I don't care too much.